lib/search/query_components/core_query.rb

```diff
@@ -126,12 +124,26 @@ def match_bigrams(fields)
       }
     end
 
+    # Use the synonym variant of the field unless we're disabling synonyms
+    def synonym_field(field_name)
+      return field_name if search_params.disable_synonyms?
+      return field_name if !search_params.synonym_b_variant?
+      raise ValueError if field_name.include?(".")
+
+      field_name + ".synonym"
+    end
+
   private
 
     def query_analyzer
       if search_params.disable_synonyms?
+        # this is the default defined in the mapping for regular fields
         DEFAULT_QUERY_ANALYZER_WITHOUT_SYNONYMS
+      elsif search_params.synonym_b_variant?
+        # this is the default defined in the mapping for *.synonym fields
+        QUERY_TIME_SYNONYMS_ANALYZER
       else
+        # this excludes the synonym filter
```

Comment:
I wrote this, but I think this should say "includes" actually

Reply:
Maybe "this includes the old synonym filter"?

In any case, we'll hopefully be getting rid of this at the end of the A/B test.